src/status_im/i18n.cljs

```diff
@@ -52,73 +53,181 @@
+                             :zh-hans-sg  zh-hans/translations
+                             :zh-hans-hk  zh-hans/translations
+                             :zh-hans-cn  zh-hans/translations
+                             :zh-hans-mo  zh-hans/translations
+                             :zh-hant     zh-hant/translations
+                             :zh-hant-tw  zh-hant/translations
+                             :zh-hant-sg  zh-hant/translations
+                             :zh-hant-hk  zh-hant/translations
+                             :zh-hant-cn  zh-hant/translations
+                             :zh-hant-mo  zh-hant/translations
+                             :zh-wuu      zh-wuu/translations
+                             :zh-yue      zh-yue/translations})
+
+;; english as source of truth
+(def labels (set (keys en/translations)))
+
+(spec/def ::label labels)
+(spec/def ::labels (spec/coll-of ::label :kind set? :into #{}))
+
+(defn labels-for-all-locales []
+  (->> translations-by-locale
+       (mapcat #(-> % val keys))
+       set))
+
+;; checkpoints
+
+;; Checkpoints specify milestones for locales.
+;;
+;; With milestones we can ensure that expected supported languages
+;; are actually supported, and visualize the translation state for
+;; the rest of locales according to these milestones.
+;;
+;; Checkpoints are defined by indicating the labels that need to be present
+;; in a locale to achieve that checkpoint.
+;;
+;; We need to define the checkpoint that needs to be achieved for
+;; a locale to be considered supported. This is why as we develop
+;; we add translations, so we need to be defining a new target
+;; for supported languages to achieve.
+;;
+;; Checkpoints are only used in dev and test. In dev when we want to
+;; manually check the state of checkpoints for locales, and in test
+;; to automatically check supported locales against the target checkpoint.
+
+(spec/def ::checkpoint.id keyword?)
+(spec/def ::checkpoint-defs (spec/map-of ::checkpoint.id ::labels))
+
+(def checkpoint-1-labels labels)
+;; these could be status versions instead of incremental numbers
+(def checkpoints-def (spec/assert ::checkpoint-defs
+                                  {::checkpoint-1 checkpoint-1-labels
+                                   ::checkpoint-2 (into checkpoint-1-labels #{})}))
+(def checkpoints (set (keys checkpoints-def)))
+
+(spec/def ::checkpoint checkpoints)
+
+(def checkpoint-to-consider-locale-supported ::checkpoint-1)
+
+(defn checkpoint->labels [checkpoint]
+  (get checkpoints-def checkpoint))
+
+(defn checkpoint-val-to-compare [c]
+  (-> c name (string/replace #"^.*\|" "") int))
+
+(defn >checkpoints [& cs]
+  (apply > (map checkpoint-val-to-compare cs)))
+
+;; locales
+
+(def locales (set (keys translations-by-locale)))
+
+(spec/def ::locale locales)
+(spec/def ::locales (spec/coll-of ::locale :kind set? :into #{}))
+
+(def supported-locales (spec/assert ::locales #{:en :es}))
```

Comment:
Please add a comment above this saying something like: `;; NOTE: Add new locale keywords here to indicate support for them`